nss/aad_auth/aadauth_test.go

```diff
@@ -0,0 +1,103 @@
+package main
+
+import (
+	"context"
+	"flag"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+	"github.com/ubuntu/aad-auth/internal/cache"
+	"github.com/ubuntu/aad-auth/internal/testutils"
+)
+
+func TestGetAllEnt(t *testing.T) {
```

Comment:
Another thing that we discussed yesterday was to take all the user cases/copy the integration tests cases basically and map them all, as they cover quite some large part of the code and I don’t see it?

Finally, we also mentioned to add a new use case with empty database, it’s not here if I am correct?

Reply:
They could be on the same test func indeed. I made them separate because the tests that only want a single entry can be executed in parallel. I thought it would be better to have this distinction... However, having to create only a single database for them would probably be faster than what we've got right now.

Reply:
good spot about parallel/non parallel.

If we want to be overkill, we can actually add a "parallel" parameter to the struct and only run t.Parallel() when needed, BUT I think this is really overkill. Let’s only gain on maintenance side.

Reply:
Not running the integration tests directly, but we mentioned yesterday running the *test cases*. Basically, we already have most of the use cases we want to test against and having both set of values tested on the CLI and integration tests helps. Like if both fails -> probably the issue is on the Go side. If only integration tests fails -> the issue is in the horrible C code :)